packages/svelte-ux/src/lib/components/theme.ts

```diff
@@ -1,25 +1,42 @@
 import type { ComponentProps, SvelteComponent } from 'svelte';
 import type * as Components from './';
-import { getSettings } from './settings';
+import { getSettings, type DefaultProps, type Settings } from './settings';
 
-type ComponentName = keyof typeof Components;
+export type ComponentName = keyof typeof Components;
 
 type ClassesProp<T> = T extends { prototype: infer PR extends SvelteComponent }
   ? ComponentProps<PR> extends { classes?: any }
     ? ComponentProps<PR>['classes']
     : never
   : never;
 
+export type ResolvedComponentClasses = {
+  [key in ComponentName]: ClassesProp<(typeof Components)[key]> extends never
+    ? {}
+    : NonNullable<ClassesProp<(typeof Components)[key]>>;
+};
+
```

Comment:
TS wasn't doing a good job of inferring the types, giving all sorts of weird errors once I tried to factor out resolveComponentClasses. This allowed me to just explicitly name the type, with the added benefit of full type info on the returned settingsClasses in each component.